plugin/CinderCompatibleAPI/api/router.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2018 Huawei Technologies Co., Ltd. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+This module implements a entry into the OpenSDS northbound REST service.
+
+*/
+
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/astaxie/beego"
+	bctx "github.com/astaxie/beego/context"
+	c "github.com/opensds/opensds/client"
+	"github.com/opensds/opensds/osdsctl/cli"
+	"github.com/opensds/opensds/pkg/api/filter/auth"
+	"github.com/opensds/opensds/pkg/api/filter/context"
+	"github.com/opensds/opensds/pkg/utils/constants"
+	"github.com/opensds/opensds/plugin/CinderCompatibleAPI/converter"
+)
+
+const (
+	// StatusOK ...
+	StatusOK = http.StatusOK
+	// StatusAccepted ...
+	StatusAccepted = http.StatusAccepted
+)
+
+var (
+	client *c.Client
+)
+
+// Run ...
+func Run(CinderEndPoint string) {
+	ep, ok := os.LookupEnv(cli.OpensdsEndpoint)
+	if !ok {
+		fmt.Println("ERROR: You must provide the endpoint by setting " +
+			"the environment variable OPENSDS_ENDPOINT")
+		return
+	}
+	cfg := &c.Config{Endpoint: ep}
+
+	authStrategy, ok := os.LookupEnv(cli.OpensdsAuthStrategy)
+	if !ok {
+		authStrategy = c.Noauth
```

Comment:
This server will run with openstack server, set the keystone as the default authentication strategy may be better.